bindsnet/network/nodes.py

```diff
@@ -703,3 +697,80 @@ def reset_(self) -> None:
         super().reset_()
         self.v = self.rest * torch.ones(self.shape)  # Neuron voltages.
         self.u = self.b * self.v                     # Neuron recovery.
+
+
+class IzhikevichMetabolicNodes(Nodes):
+    # language=rst
+    """
+    Layer of Metabolic Izhikevich neurons.
+    """
+
+    def __init__(self, n: Optional[int] = None, shape: Optional[Iterable[int]] = None, traces: bool = False,
+                 excitatory: bool = True, thresh: float = -52.0, rest: float = -65.0, 
+				 trace_tc: float = 5e-2, beta: float = 0.5) -> None:
+        # language=rst
+        """
+        Instantiates a layer of Izhikevich neurons.
+
+        :param n: The number of neurons in the layer.
+        :param shape: The dimensionality of the layer.
+        :param traces: Whether to record spike traces.
+        :param excitatory: Whether layer is excitatory.
+        :param thresh: Spike threshold voltage.
+        :param rest: Resting membrane voltage.
+        :param trace_tc: Time constant of spike trace decay.
+        :param beta: beta_input of the metabolic model.
+        """
+        super().__init__(n, shape, traces, trace_tc)
+
+        self.rest = rest       # Rest voltage.
+        self.thresh = thresh   # Spike threshold voltage.
+        self.beta = beta
+
+        if excitatory:
+            self.r = torch.rand(n)
+            self.a = 0.02 * torch.ones(n)
+            self.b = 0.2 * torch.ones(n)
+            self.c = -65.0 + 15 * (self.r ** 2)
+            self.d = 8 - 6 * (self.r ** 2)
+        else:
+            self.r = torch.rand(n)
+            self.a = 0.02 + 0.08 * self.r
+            self.b = 0.25 - 0.05 * self.r
+            self.c = -65.0 * torch.ones(n)
+            self.d = 2 * torch.ones(n)
+
+        m = 0.7
```

Comment:
Should be `self.m = 0.7`; otherwise, it can't be accessed elsewhere.

Reply:
You right